File: Data Structures and Algorithms/DSA-in-C-/algo/sorting/insertionSort/insertionSort.cpp
#include <iostream>

using std::swap;

bool isSorted(int arr[], int size){
  //Let's check if the array only has 1 element on it. if it is then its true
  //Same goes if the size is 0
  if(size == 1 || size == 0)
    return true;
  for(int i=1; i<size; i++)
    if(arr[i-1] < arr[i]) //Looks for unsorted pair in the array
      return false;   //If there is then it returns false

  return true;
}

//Prints array function
void print(int arr[], int size){
    for(int i=0; i<size; i++)
        std::cout<<"[ "<<arr[i]<<"] ";
}

//Insertion sort function
void insertionSort(int arr[], int size){
    int i,min,j;
    
    for(i=1; i<size; i++){
        min = arr[i];
        j = i - 1;
        while(j >= 0 && arr[j] > min){
            arr[j+1] = arr[j];
            j = j-1;
        }
        arr[j+1]=min;
}
}

int main(){
    int array[10] = {1,2,3,4,5,6,7,8,9,10};
    int size = sizeof(array) / sizeof(array[0]);
    if(!isSorted(array,size)){
        std::cout<<"There's nothing to sort tho :) \n";
    }
    else
        insertionSort(array,size);

        print(array,size);

        return 0;
}

File: Data Structures and Algorithms/DS-A-in-Java/algo/sort/insertionSort/insertionSort.java
import java.util.Scanner;

class insertionSort{
/*
Made by: Sean Christian Lozana
*/
	private boolean isSorted(int[] arr){
		//Checks if the size of the array are base-cases if its 0 or 1 then its a sorted case
		if(arr.length == 1 || arr.length == 0)
			return true;
		//Traverses the array in a linear fashion to check if there is an unsorted pair
		//If true then return false
		for(int i=1; i<arr.length; i++)
			if(arr[i] < arr[i+1])
				return false;
	//If the array doesn't have an unsorted pair then return true the array is sorted
		return true;
	}

	//Prints the array
	private void print(int[] arr){
		for(int i=0; i<arr.length; i++)
			System.out.print("["+arr[i]+"] ");
	}


	private void toInsertionSort(int[] arr){
		int i, min, j;
		for (i = 1; i < arr.length; i++) 
    {  
        min = arr[i]; //element at arr[i] gets assigned at min
        j = i - 1; 
        while (j >= 0 && arr[j] > min) 
        {  
            arr[j + 1] = arr[j];  
            j = j - 1;
        }  
        arr[j + 1] = min;
    } 
	}

	public static void main(String[] args) {
	Scanner input = new Scanner(System.in);
	System.out.print("Enter array size: ");
	int size = input.nextInt();
	int[] array = new int[size];
	System.out.print("Enter ["+size+"] numbers: ");
	for(int i=0; i<array.length; i++)
		array[i] = input.nextInt();
	System.out.print("Unsorted array: ");
	new insertionSort().print(array);
	System.out.print("\nSorted array: ");
	new insertionSort().toInsertionSort(array);
	new insertionSort().print(array);
	}
}